Fix release script not triggering action to publish in PyPI (#331)
appliedAI-Initiative/pyDVL: CONTRIBUTING.md

2. We try not to trigger CI pipelines when unnecessary (see [Skipping CI
runs](#skipping-ci-runs)).

### Running Github Actions locally

To run Github Actions locally we use [act](https://github.com/nektos/act).
It uses the workflows defined in `.github/workflows` and determines
the set of actions that need to be run. It uses the Docker API
to either pull or build the necessary images, as defined
in our workflow files and finally determines the execution path
based on the dependencies that were defined.

Once it has the execution path, it then uses the Docker API
to run containers for each action based on the images prepared earlier.
The [environment variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) 

> and [filesystem](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#file-systems) are all configured to match
> what GitHub provides.

Reviewer: ```suggestion
and [filesystem](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#file-systems)
are all configured to match what GitHub provides.
```

```shell
curl -s https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash -s -- -d -b ~/bin 
```

And then simply add it to your PATH variable: `PATH=~/bin:$PATH`

Refer to its official
[readme](https://github.com/nektos/act#installation-through-package-managers)
for more installation options.

#### Cheatsheat

```shell
# List all actions for all events:
act -l

# List the actions for a specific event:
act workflow_dispatch -l

# List the actions for a specific job:
act -j lint -l

# Run the default (`push`) event:
act

# Run a specific event:
act pull_request

# Run a specific job:
act -j lint

# Collect artifacts to the /tmp/artifacts folder:
act --artifact-server-path /tmp/artifacts

# Run a job in a specific workflow (useful if you have duplicate job names)
act -j lint -W .github/workflows/tox.yml

# Run in dry-run mode:
act -n

# Enable verbose-logging (can be used with any of the above commands)
act -v
```

#### Example

To run the `publish` job (the toughest one to test) with tag 'v0.6.0' 
you would simply use:

```shell
act push -j publish --eventpath events.json
```

With `events.json` containing:

```json
{
  "ref": "refs/tags/v0.6.0"
}
```

> To instead run it as it were manually triggered (i.e. `workflow_dispatch`)

Reviewer: ```suggestion
To instead run it as if it had been manually triggered (i.e. `workflow_dispatch`)

```

workflow to publish packages to [PyPI](https://pypi.org/) from `develop` after
a GitHub release.

#### Publish to TestPyPI

We use [bump2version](https://pypi.org/project/bump2version/) to bump

> the build part of the version number publish a package to TestPyPI from CI.

Reviewer: ```suggestion
the build part of the version number and publish a package to TestPyPI from CI.
```
